front_matter/any.ts

```diff
@@ -34,7 +34,9 @@ export type { Extract };
  * @param text The text to extract front matter from.
  * @returns The extracted front matter and body content.
  */
-export function extract<T>(text: string): Extract<T> {
+export function extract<T extends Record<string, unknown>>(
+  text: string,
+): Extract<T> {
```

Comment:
Can we keep the existing type? `T` can be, for example, boolean. This change breaks that usage:

```js
import { extract } from "@std/front-matter/any";

console.log(extract<boolean>(`---
true
---

aaa
`));
```